lib/Doctrine/DBAL/Schema/Table.php

```diff
@@ -634,11 +634,7 @@ public function getColumns()
         }
         $colNames = array_unique(array_merge($pkCols, $fkCols, array_keys($columns)));
 
-        uksort($columns, function ($a, $b) use ($colNames) {
-            return (array_search($a, $colNames) >= array_search($b, $colNames));
-        });
-
-        return $columns;
+        return array_replace(array_flip($colNames), $columns);
```

Comment:
For some reasons PK columns are put first, the FK columns, then the other ones. No idea why that was implemented like this. No idea if anyone relies on that...